### PR TITLE
fix(#119): add new home for configuration files

### PR DESCRIPTION
**Pull request by dipak-pawar**
Fixes: #119

### CODE REVIEW

**Comment by alien-ike** [comment]
### Ike Plugins (test-keeper)

Thank you @dipak-pawar for this contribution!



It appears that no tests have been added or updated in this PR.

Automated tests give us confidence in shipping reliable software. Please add some as part of this change.

If you are an admin or the reviewer of this PR and you are sure that no test is needed then you can use the command `/ok-without-tests` as a comment to make the status green.


Your plugin configuration is stored in the [file](https://github.com/arquillian/ike-prow-plugins/cb93dba0df1a9db0c2751573dfc6fc83d5443488/test-keeper.yaml).

**Comment by MatousJobanek** [comment]
Thanks @dipak-pawar for your work, unfortunately, I'm afraid that the task was misunderstood. The new default location of the config files `.ike-prow/` should be something that is automatically checked for the presence of the files - for both the configuration file and plugin hint file (with that fact that the location of the plugin hint file can be still customized in config file). It should work in the same way as the `.github` directory works for the GH templates.
The `pkg/assets/config/test-keeper.yaml` should stay there where it was - this contains default inclusions and exclusions applicable for all projects the plugins are used in. The `.ike-prow/test-keeper.yaml` file contains additional configuration specific for the particular project - our own one is here: https://github.com/arquillian/ike-prow-plugins/blob/master/test-keeper.yaml that one should have been moved to the directory.

**Comment by MatousJobanek** [comment]
[test]

**Comment by dipak-pawar** [comment]
@MatousJobanek can you look into this when you find some time?

File: pkg/assets/config/test-keeper.yaml [hunk 1]
```diff
@@ -134,3 +135,4 @@ skip_validation_for:
 
   # Our own configurations
   - 'regex{{test-keeper\.y[a]?ml$}}'
```
Comment (MatousJobanek) [inline pkg/assets/config/test-keeper.yaml]: is this line still necessary when the configs should be inside of the directory that is excluded?

File: .ike-prow/test-keeper.yaml [hunk 2]
```diff
@@ -0,0 +1,2 @@
+skip_validation_for: 
+  - 'cluster/'
```
Comment (MatousJobanek) [inline .ike-prow/test-keeper.yaml]: there is still also the original file https://github.com/dipak-pawar/ike-prow-plugins/blob/c32f52a822cedb073f21298dda3e98ecb424bf82/test-keeper.yaml - it should be removed (moved here)

File: pkg/github/service/config_loader.go [hunk 3]
```diff
@@ -44,6 +44,16 @@ func (l *LoadableConfig) loadFromRawFile(pathTemplate string) config.Source {
...
 		l.BaseConfig.PluginName = l.PluginName
+
+		pluginHintPath := fmt.Sprintf(".ike-prow/%s_hint.md", l.PluginName)
+		hintURL := rawFileService.GetRawFileURL(pluginHintPath)
+
+		_, e := utils.GetFileFromURL(hintURL)
+
+		if e == nil {
+			l.BaseConfig.PluginHint = githubBaseURL + rawFileService.GetRelativePath(pluginHintPath)
+		}
```
Comment (MatousJobanek) [inline pkg/github/service/config_loader.go]: this shouldn't be here. This function is for loading configuration file, not the *_hint file. 
In addition, on the line 51 you download the file - that should be lazy. Download it (and generate the default path) only when needed. 
This should be part of `comment_message.go`

File: pkg/plugin/test-keeper/configuration_gh_test.go [hunk 4]
```diff
@@ -24,8 +24,10 @@ var _ = Describe("Test keeper config loader features", func() {
 
 		It("should load test-keeper configuration yml file", func() {
 			// given
+			NonExistingRawGitHubFiles(".ike-prow/test-keeper_hint.md")
```
Comment (MatousJobanek) [inline pkg/plugin/test-keeper/configuration_gh_test.go]: I don't think that this is necessary as it checks the suffix

**Comment by MatousJobanek** [comment]
- [ ] Please also change the documentation to explain the "workflow" of loading the `*_hint.md` files - that when nothing is set in config, then it is automatically loaded from the pre-defined location.

**Comment by MatousJobanek** [comment]
@dipak-pawar  please, take a look at my last changes - if it makes sense to you and you are fine with it.

**Comment by dipak-pawar** [comment]
@MatousJobanek Thanks for all changes. lgtm

**Comment by MatousJobanek** [comment]
[test]

**Pull request merged**